doc/developer/WritingEndToEndTests.adoc

```diff
@@ -0,0 +1,109 @@
+
+=== Folder & Files architecture
+
+The End-to-End tests are located in `<ARA_ROOT>/client/cypress`. It contains the following hierarchy :
+
+* `plugins` : Use for integration of plugins for cypress (like the `cypress-cucumber-preprocessor` one).
+* `scenarii` : Stores all the gherkin `.feature` files. We can freely create sub-directories in it if you want.
+* `support` : implementations of tests
+** `support/fragments` : Contains all the Page objects and modules. The name  of each file should depict correctly the
+page it represents.
+** `support/stepDefinitions` : link between Gherkin `.feature` and Javascript code. Ideally, one file should represent
+one page.
+** `support/commands.js` : Global commands and overloads of Cypress module
+** `support/index.js` : The entry point of Cypress
+
+== How to launch the tests
+
+You can either launch all the end to end tests with the command : `npm run cy:all`.
+
+Or you can add the `@wip` tag on scenario you choose in the `.feature` files and run them with `npm run cy:wip`.
+
+=== Create a specialized runner
+
+If you want to run only the scenarios with a specific tag (like `@severity-sanity-check`), you can update the
+`<ARA_ROOT>/client/package.json` with a new target (in the `scripts` part) which will have the following syntax :
+
+`"cy:<name>": cypress run -e TAGS='<your_tag>' -e GLOB=test/cypress/**/*.feature`
+
+With :
+* `name` the name of your target (like `wip` or `sanity` for example)
+* `your_tag` the tag your target (like `@wip` or `@severity-sanity-check`)
+
+And then, you can run it with `npm run cy:<name>`.
+
+== How to write a new tests for ARA
+
+=== Create the Gherkin scenario
+
+First you can create your scenario in the `test/cypress/scenarii` folder (either create a new file or add it to an
+existing one). Don't hesitate to see the other `.feature` files to check what are the available sentences, but you can
+create a sentence if you don't find the one you're looking for.
+
+The only required thing is to respect the Gherkin syntax (Given/When/Then).
+
+=== Target Frontend components
+
+If you need to target specific front end components in your Gherkin sentences, you can locate them (quicktip : Find a
+specific label in ARA, then search it in the client sources), and add the custom attribute `data-nrt="<your-id>"` in it.
+
+In your Javascript code, you'll retrieve the element with the code `cy.GetByDataNrt("<your-id>")` or
+`cy.WaitForElementByDataNrt("<your-id>", timeoutValueInMs)` if your element appears dynamically.
+
+You can check the `<ARA_ROOT>/client/src/views/management-projects.vue` file for an example.
+
+=== Implement the Gherkin sentences
+
+If you create your own sentences, then you'll need to implement the behavior of it with some javascript code.
+
+Don't hesitate to get inspirations from already existing code in `test/cypress/support`.
+NOTE: You must known the base of javascript and NodeJS basics to create End-To-End tests.
```

Comment:
`You must know the basics of Javascript and NodeJS to create End-to-End tests` instead of `You must known the base of javascript and NodeJS basics to create End-To-End tests`